harmonica/meshes.py

```diff
@@ -0,0 +1,212 @@
+
+    Pass top and bottom as ellipsoidal heights (geodetic coordinate system)
+
+    >>> bottom, top = -1000, 1000
+    >>> region = [-12, 12, -20, 20]
+    >>> layer = tesseroids_layer(
+    ...     region, spacing=0.5, top=top, bottom=bottom, coordinates="geodetic"
+    ... )
+    >>> print(layer.data_vars)
+    Data variables:
+        top      (latitude, longitude) float64 6.377e+06 6.377e+06 ... 6.377e+06
+        bottom   (latitude, longitude) float64 6.375e+06 6.375e+06 ... 6.375e+06
+        density  (latitude, longitude) float64 0.0 0.0 0.0 0.0 ... 0.0 0.0 0.0 0.0
+
+    Create a layer where region coordinates are the boundaries of extreme tesseroids
+
+    >>> region = [-1, 1, -1, 1]
+    >>> layer = tesseroids_layer(region, spacing=0.2, region_centers=False)
+    >>> print(layer.coords)
+    Coordinates:
+      * longitude  (longitude) float64 -0.9 -0.7 -0.5 -0.3 -0.1 0.1 0.3 0.5 0.7 0.9
+      * latitude   (latitude) float64 -0.9 -0.7 -0.5 -0.3 -0.1 0.1 0.3 0.5 0.7 0.9
+
+    """
+    check_region(region)
+    if shape is not None and spacing is not None:
+        raise ValueError("Both shape and spacing provided. Only one is allowed.")
+    if shape is None and spacing is None:
+        raise ValueError("Either a shape or a spacing must be provided.")
+    if spacing is not None:
+        shape, region = spacing_to_shape(region, spacing, adjust)
+    if coordinates not in ("spherical", "geodetic"):
+        raise ValueError(
+            "Invalid geographic coordinate system '{}'.".format(coordinates)
+            + " Should be 'spherical' or 'geodetic'."
+        )
+    longitude = np.linspace(region[0], region[1], shape[1])
+    latitude = np.linspace(region[2], region[3], shape[0])
+    d_lon, d_lat = longitude[1] - longitude[0], latitude[1] - latitude[0]
+    if not region_centers:
+        longitude = longitude[:-1] + d_lon / 2
+        latitude = latitude[:-1] + d_lat / 2
+        shape = (shape[0] - 1, shape[1] - 1)
+    coords = {"longitude": longitude, "latitude": latitude}
+    if top is None:
+        top = np.nan
+    if bottom is None:
+        bottom = np.nan
+    if density is None:
+        density = 0
+    dims = ("latitude", "longitude")
+    data_vars = {
+        "top": (dims, top * np.ones(shape)),
+        "bottom": (dims, bottom * np.ones(shape)),
+        "density": (dims, density * np.ones(shape)),
+    }
+    metadata = {"spacing": (d_lat, d_lon), "region": region, "shape": shape}
+    layer = xr.Dataset(data_vars, coords=coords, attrs=metadata)
+    # Convert top and bottom coordinates if given in geodetic
+    if coordinates == "geodetic":
+        if top is not None:
+            _, _, layer["top"] = geodetic_to_spherical(
+                layer.longitude, layer.latitude, layer.top
+            )
+        if bottom is not None:
+            _, _, layer["bottom"] = geodetic_to_spherical(
+                layer.longitude, layer.latitude, layer.bottom
+            )
```

Comment:
What about the latitudes? It would make more sense for everything to be in geodetic, not just top and bottom. It's OK if the grid is not evenly spaced in spherical latitude if you ask for geodetic coordinates.

Reply:
I left the latitudes unchanged because I want to have evenly spaced grids on spherical coordinates.
Should we handle this differently?

Maybe the docstring is not clear enough, but the `tesseroid_layer` will always return the xarray in spherical coordinates. So when I pass `coordinates="geodetic"` it means that the `top` and `bottom` are in geodetic coordinates, but the returned xarray will be in spherical.